cmd/virtctl/virtctl.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/gorilla/websocket"
+	"golang.org/x/crypto/ssh/terminal"
+	"log"
+	"net/url"
+	"os"
+	"os/signal"
+	"time"
+)
+
+func main() {
+
+	var addr = flag.String("master", "localhost:8080", "http service address")
+	var vm = flag.String("vm", "", "VM to connecto to")
+	var device = flag.String("device", "serial0", "Serial device to connect to")
+	var namespace = flag.String("namespace", "default", "Namespace of the VM")
+
+	flag.Parse()
+	log.SetFlags(0)
+	log.SetOutput(os.Stderr)
+
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt)
+
+	u := url.URL{Scheme: "ws", Host: *addr, Path: fmt.Sprintf("/apis/kubevirt.io/v1alpha1/namespaces/%s/vms/%s/console", *namespace, *vm)}
+	u.RawQuery = "console=" + *device
+	log.Printf("connecting to %s", u.String())
+
+	c, _, err := websocket.DefaultDialer.Dial(u.String(), nil)
+	if err != nil {
+		log.Fatal("dial:", err)
+	}
+	defer c.Close()
+
+	writeStop := make(chan struct{})
+	readStop := make(chan struct{})
+
+	state, err := terminal.MakeRaw(int(os.Stdin.Fd()))
+	if err != nil {
+		log.Fatal("Make raw terminal:", err)
+	}
+	defer terminal.Restore(int(os.Stdin.Fd()), state)
+	fmt.Fprint(os.Stderr, "Escape sequence is ^]")
+
+	go func() {
+		defer close(readStop)
+		for {
+			_, message, err := c.ReadMessage()
```

Comment:
This looks like it's expecting an "eof" type of error, but c is a general DefaultDialer. Are there any other error conditions to be worried about?